### ISSUE TITLE
Suggestion: Separate tree building and output generation

### ISSUE THREAD

**Issue by tehbilly** [analog-nico/express-print-routes#1]
I'd love to give this a whirl, but the "only prints to a file" bit is a touch limiting.

My thoughts would be to separate tree building and output. A way to get the tree as a string, an object *(for serialization or using as an API endpoint. Recursion!)*, or pass to my own writer would be my ideal.

**Comment by analog-nico** [comment]
Interesting use cases! I'll do some refactoring and expose internal plumbing to accommodate them.

**Comment by jasonkarns** [comment]
Agreed. I would expect this to be a CLI that would emit the tree to STDOUT where I could redirect to a file if I desired (or to other pieces in my build process).

**Comment by jasonkarns** [comment]
Further utility for this would be exposing a route on the app itself such that one could hit the running app and get a print out of the route tree (as Rails provides).

**Comment by analog-nico** [comment]
@jasonkarns It cannot become a CLI since it requires the express app to be up and running. A CLI would need to do some kind of static code analysis and that is a whole different story.

To the point of this issue I think a callback would be useful:

``` js
// Either writing to a file:
require('express-print-routes')(app, filepath)

// Or doing whatever you want with the JSON object:
require('express-print-routes')(app, function (jsonTree) {
    // Register another API endpoint, write to cache etc.
})

**Comment by jasonkarns** [comment]
I'd have the CLI take in the file of the app startup (startApp.js or server.js or similar) and spin it up like usual. It wouldn't be static, you'd be spinning it up, mounting and shutting down.
